flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/HsResultPartitionReadSchedulerTest.java

```diff
@@ -0,0 +1,403 @@
+
+        assertThat(ioExecutor.numQueuedRunnables()).isZero();
+    }
+
+    /** Test scheduler was released, but receive new subpartition reader registration. */
+    @Test
+    void testRegisterSubpartitionReaderAfterSchedulerReleased() {
+        TestingHsSubpartitionFileReader.Factory factory =
+                new TestingHsSubpartitionFileReader.Factory();
+        readScheduler.release();
+        assertThatThrownBy(
+                        () -> {
+                            readScheduler.registerNewSubpartition(
+                                    0, subpartitionViewOperation, factory);
+                            ioExecutor.trigger();
+                        })
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessageContaining("HsResultPartitionReadScheduler is already released.");
+    }
+
+    private static FileChannel openFileChannel(Path path) throws IOException {
+        return FileChannel.open(path, StandardOpenOption.READ);
+    }
+
+    private static class TestingHsSubpartitionFileReader implements SubpartitionFileReader {
+        private final Runnable prepareForSchedulingRunnable;
+
+        private final BiConsumerWithException<Queue<MemorySegment>, BufferRecycler, IOException>
+                readBuffersConsumer;
+
+        private final Consumer<Throwable> failConsumer;
+
+        private final Supplier<Integer> prioritySupplier;
+
+        public TestingHsSubpartitionFileReader(
+                Runnable prepareForSchedulingRunnable,
+                BiConsumerWithException<Queue<MemorySegment>, BufferRecycler, IOException>
+                        readBuffersConsumer,
+                Consumer<Throwable> failConsumer,
+                Supplier<Integer> prioritySupplier) {
+            this.prepareForSchedulingRunnable = prepareForSchedulingRunnable;
+            this.readBuffersConsumer = readBuffersConsumer;
+            this.failConsumer = failConsumer;
+            this.prioritySupplier = prioritySupplier;
+        }
+
+        @Override
+        public void prepareForScheduling() {
+            prepareForSchedulingRunnable.run();
+        }
+
+        @Override
+        public void readBuffers(Queue<MemorySegment> buffers, BufferRecycler recycler)
+                throws IOException {
+            readBuffersConsumer.accept(buffers, recycler);
+        }
+
+        @Override
+        public void fail(Throwable failureCause) {
+            failConsumer.accept(failureCause);
+        }
+
+        @Override
+        public int compareTo(SubpartitionFileReader that) {
+            checkArgument(that instanceof TestingHsSubpartitionFileReader);
+            return Integer.compare(
+                    prioritySupplier.get(),
+                    ((TestingHsSubpartitionFileReader) that).prioritySupplier.get());
+        }
+
+        public static class Factory implements SubpartitionFileReader.Factory {
```

Comment:
It's probably easier for this factory to just poll from a queue of readers on each `createFileReader`. E.g., we can construct `TestingHsSubpartitionFileReader` with a priority that is used for comparing w/o a priority supplier.

Reply:
good suggesstion.